providers/flagd/README.md

```diff
@@ -134,14 +135,30 @@ platforms or architectures.
 Reconnection is supported by the underlying GRPCBlockingStub. If the connection to flagd is lost, it will reconnect
 automatically.
 
-### Deadline (gRPC call timeout)
+### Deadlines
+
+Deadlines are used to define how long the provider waits to complete initialization or flag evaluations. 
+They behave differently based on the resolver type.
+
+#### Deadlines with Remote resolver (RPC)
 
 The deadline for an individual flag evaluation can be configured by calling `setDeadline(myDeadlineMillis)`.
-If the gRPC call is not completed within this deadline, the gRPC call is terminated with the error `DEADLINE_EXCEEDED`
+If the remote evaluation call is not completed within this deadline, the gRPC call is terminated with the error `DEADLINE_EXCEEDED`
 and the evaluation will default.
-The default deadline is 500ms, though evaluations typically take on the order of 10ms.
-For the in-process provider, the deadline is used when establishing the initial streaming connection.
-A failure to connect within this timeout will result in an [error event](https://openfeature.dev/docs/reference/concepts/events#provider_error) from the provider, though it will attempt to reconnect indefinitely.
+
+#### Deadlines with In-process resolver
+
+In-process resolver with remote evaluation uses the `deadline` for synchronous gRPC calls to fetch metadata from flagD.
+If fetching metadata fails within this deadline, the provider will try to reconnect.
+The `streamDeadlineMs` defines a deadline for the streaming connection that listens to flag configuration updates from 
+flagD. After the deadline exceeds, the provider closes the gRPC stream and will attempt to reconnect.
+
+A failure to connect within the deadline will result in
+an [error event](https://openfeature.dev/docs/reference/concepts/events#provider_error) from the provider, though it will attempt to reconnect indefinitely.
```

Comment:
```suggestion
A failure to connect will result in
an [error event](https://openfeature.dev/docs/reference/concepts/events#provider_error) from the provider, though it will attempt to reconnect indefinitely.
```

I don't this comment is entirely correct, or at least it's not fully comprehensive; it makes it sound like the connection will wait 10 minutes to error (which it will not) if the underlying channel can't be opened, it will error immediately when the stream is attempted and then go into the backoff/retry.

Reply:
I moved this paragraph to the [reconnection section](https://github.com/open-feature/java-sdk-contrib/pull/1011/files#diff-04582b0607c583d36a31ea656e9229beceab5f2abc3139710831765da374ce89R137-R138).